app/posts/releasing-rails-libraries-for-version-5.md

```diff
@@ -0,0 +1,131 @@
+        text: "Identity check",
+        hint: <<~HINT
+          We need your personal and contact details in order
+          to begin confirming your identity
+        HINT
+      )
+      item.with_status(text: "Cannot start")
+    end
+  end
+%>
+```
+
+The above snippet will render this task list:
+
+![A GOV.UK design system task list with entries for personal and contact details, and a greyed out identity check task that cannot yet be started](/assets/posts/releasing-rails-libraries-for-version-5/rendered-task-list.png)
+
+## New link and button helpers
+
+The govuk-components link and button helpers are probably the most widely used feature offered by the gem.
+
+Their role is to take `some text` and `/a-link` and generate a link with design system classes like `govuk-link`, so:
+
+```erb
+<%= govuk_link_to("some text", "/a-link") %>
+
+<!-- or -->
+
+<%= govuk_link_to("/a-link") do %>
+  some text
+<% end %>
+
+<!-- becomes -->
+
+<a class="govuk-link" href="/a-link">some text</a>
+```
+
+In November 2022 [I tried to add a new feature](https://github.com/x-govuk/govuk-components/pull/363) to `govuk_link_to` which allowed developers to call it with `new_tab: true`, and the resulting link would open pages in a new tab.
+
+Unfortunately [I had to revert the change shortly afterwards](https://github.com/x-govuk/govuk-components/pull/399) because mixing new Ruby positional keyword arguments (added in 2019) with old Rails link helpers (unchanged since 2004) caused lots of problems.
+
+Thankfully we uncovered the problems in our pre-release testing so didn't break anything important.
+
+Still, opening links in a new tab was something I wanted to address. It's more complicated than the other modifiers like `no_underline` because it adds multiple new attributes to links. Developers often forget the `rel` attribute, leaving users open to [reverse tabnabbing](https://owasp.org/www-community/attacks/Reverse_Tabnabbing). We can provide real value for users by supporting this feature.
+
+In the new version, all of the design system options for links and buttons are represented by keyword arguments. They allow us to [support a large number of parameters](https://govuk-components.netlify.app/helpers/link/) while keeping [the code](https://github.com/x-govuk/govuk-components/blob/main/app/helpers/govuk_link_helper.rb) clean and succinct.
+
+Now developers can reliably build complex link markup with confidence:
+
+```erb
+<%= govuk_link_to(
+  "View",
+  "/my-profile",
+  visually_hidden_suffix: "my profile",
+  new_tab: true,
+  no_visited_state: true
+) %>
+
+<!-- becomes -->
+
+<a class="govuk-link govuk-link--no-visited-state" target="_blank" rel="noreferrer noopener">
+  View<span class="govuk-visually-hidden"> my profile</span>
+</a>
+```
+
+## What's next
+
+Now version 5 is released we'll slow the pace for a while and let the dust settle. This should give developers the opportunity to try out the new features and provide feedback or report bugs.
```

Comment:
Make this slightly more direct? Should we say how to report bugs/give feedback (ie link to GitHub issues?)

```suggestion
Now version 5 is released, we'll slow the pace for a while and let the dust settle to give you the opportunity to try out the new features.

Please report any bugs, and give us your feedback.
```